content/c/concepts/linked-list/linked-list.md

```diff
@@ -0,0 +1,325 @@
+---
+Title: 'Linked List'
+Description: 'A data structure that contains a collection of nodes that together with the help of pointers, display a sequence.'
+Subjects:
+  - 'List Types'
+  - 'Computer Science'
+Tags:
+  - 'Singly linked lists'
+  - 'Doubly linked lists'  
+  - 'Circular linked lists'
+  - 'Circular doubly linked lists'
+CatalogContent:
+  - 'learn-c'
+  - 'paths/computer-science'
+---
+
+A **Linked List** is a data structure that contains a collection of nodes that together with the help of pointers, display a sequence. There are a number of list types, we will cover four of the most commmon with examples in the C language.
+
+## Syntax
+
+```pseudo
+#include 
+#include 
+
+typedef struct Node {
+    void* data;
+    struct Node *next;
+} Node;
+//push to the front of the list
+struct Node* push(struct Node* head, void* data) {
+    struct Node* new_node = (struct Node*)malloc(sizeof(struct Node));
+    new_node->data = data;
+    new_node->next = head;
+    return new_node;
+}
+int main(){
+    Node* list = NULL;
+    //remember to set the list after push
+    list = push(list, "!");
+    list = push(list, "World ");
+    list = push(list, "Hello ");
+    //print out the elements
+    Node* crnt = list;
+    while(crnt != NULL){
+        printf("%s", crnt->data);
+        crnt = crnt->next;
+    }
+    return 0;
+}
+```
+
+## Example
+
```

Comment:
Give a small introductory sentence here which gives a rough idea of what one can expect from the given code snippet present below.